refinery/core/test_views.py

```diff
@@ -499,9 +554,162 @@ def test_is_filtered_data_set_returns_false_for_public_group(self):
+        view_set.current_site = SimpleLazyObject(lambda: 'test_site')
+        email = view_set.send_tranfer_notification_email(self.user,
+                                                         new_owner, groups)
+        self.assertIn(self.user.username, email.body)
+        self.assertIn(new_owner.username, email.body)
+
+    def test_send_tranfer_notification_email_sends_profiles(self):
+        new_owner_email = 'new_owner@fake.com'
+        new_owner = User.objects.create_user('NewOwner1',
+                                             new_owner_email,
+                                             self.password)
+        groups = {'group_with_access': [], 'group_without_access': []}
+        view_set = DataSetsViewSet()
+        view_set.request = self.factory.get(self.url_root)
+        view_set.request.user = SimpleLazyObject(lambda: self.user)
+        view_set.data_set = SimpleLazyObject(lambda: self.data_set)
+        view_set.current_site = SimpleLazyObject(lambda: 'test_site')
+        email = view_set.send_tranfer_notification_email(self.user,
+                                                         new_owner,
+                                                         groups)
+        self.assertIn(
+            'http://{}/users/{}'.format(
+                view_set.current_site, self.user.profile.uuid
+            ),
+            email.body)
+        self.assertIn(
+            'http://{}/users/{}'.format(
+                view_set.current_site, new_owner.profile.uuid
+            ),
+            email.body)
+
+    def test_send_tranfer_notification_email_sends_data_set(self):
+        new_owner_email = 'new_owner@fake.com'
+        new_owner = User.objects.create_user('NewOwner1', new_owner_email,
+                                             self.password)
+        groups = {'group_with_access': [], 'group_without_access': []}
+        view_set = DataSetsViewSet()
+        view_set.request = self.factory.get(self.url_root)
+        view_set.request.user = SimpleLazyObject(lambda: self.user)
+        view_set.data_set = SimpleLazyObject(lambda: self.data_set)
+        view_set.current_site = SimpleLazyObject(lambda: 'test_site')
+        email = view_set.send_tranfer_notification_email(self.user,
+                                                         new_owner, groups)
+        self.assertIn(self.data_set.name, email.body)
+        self.assertIn(self.data_set.uuid, email.body)
+
+    def test_update_group_perms_remove_access(self):
+        new_owner_email = 'new_owner@fake.com'
+        new_owner = User.objects.create_user('NewOwner1', new_owner_email,
+                                             self.password)
+        self.data_set.share(ExtendedGroup.objects.public_group())
+        group_union = ExtendedGroup.objects.create(name="Group Union",
+                                                   is_public=True)
+        group_non_union = ExtendedGroup.objects.create(name="Group Non-Union",
```

Comment:
I believe in Refinery there is only to be one public group. Maybe this isn't testing an exact scenario we'd encounter

Reply:
Unsure what you mean. The data set is shared with three groups (one being public) and the new owner only belongs to public and the union group. This test checks to see the data set is unshared with the non-union group.